go-controller/pkg/ovn/egressip.go

```diff
@@ -412,61 +412,115 @@ func (e *egressIPMode) needsRetry(pod *kapi.Pod) bool {
+		}
+		policyIDs, err := findLRPolicyIDs(filterOption, egressIPName)
+		if err != nil {
+			return err
+		}
+		if policyIDs == "" {
+			if packetMark != 0 {
+				_, stderr, err = util.RunOVNNbctl(
+					"--id=@lr-policy",
+					"create",
+					"logical_router_policy",
+					"action=reroute",
+					fmt.Sprintf("match=\"%s\"", filterOption),
+					fmt.Sprintf("priority=%v", egressIPReroutePriority),
+					fmt.Sprintf("nexthop=%s", gatewayRouterIP),
+					fmt.Sprintf("external_ids:name=%s", egressIPName),
+					fmt.Sprintf("pkt_mark=%v", packetMark),
+					"--",
+					"add",
+					"logical_router",
+					ovnClusterRouter,
+					"policies",
+					"@lr-policy",
+				)
 			} else {
-				_, stderr, err = util.RunOVNNbctl("lr-policy-add", ovnClusterRouter, egressIPReroutePriority,
-					fmt.Sprintf("ip4.src == %s && ip4.dst == 0.0.0.0/0", podIP.String()), "reroute", gatewayRouterIP.String(), fmt.Sprintf("pkt_mark=%v", packetMark))
+				_, stderr, err = util.RunOVNNbctl(
+					"--id=@lr-policy",
+					"create",
+					"logical_router_policy",
+					"action=reroute",
+					fmt.Sprintf("match=\"%s\"", filterOption),
+					fmt.Sprintf("priority=%v", egressIPReroutePriority),
+					fmt.Sprintf("nexthop=%s", gatewayRouterIP),
+					fmt.Sprintf("external_ids:name=%s", egressIPName),
+					"--",
+					"add",
+					"logical_router",
+					ovnClusterRouter,
+					"policies",
+					"@lr-policy",
+				)
 			}
-			if err != nil && !strings.Contains(stderr, policyAlreadyExistsMsg) {
-				return fmt.Errorf("unable to create logical router policy for egress IP: %s, stderr: %s, err: %v", status.EgressIP, stderr, err)
+			if err != nil {
+				return fmt.Errorf("unable to create logical router policy: %s, stderr: %s, err: %v", status.EgressIP, stderr, err)
 			}
 		}
 	}
 	return nil
 }
 
-func (e *egressIPMode) deleteEgressPolicy(podIps []net.IP, status egressipv1.EgressIPStatusItem) error {
+func (e *egressIPMode) deleteEgressPolicy(podIps []net.IP, status egressipv1.EgressIPStatusItem, egressIPName string) error {
 	for _, podIP := range podIps {
+		var filterOption string
 		if utilnet.IsIPv6(podIP) && utilnet.IsIPv6String(status.EgressIP) {
-			_, _, err := util.RunOVNNbctl("lr-policy-del", ovnClusterRouter, egressIPReroutePriority, fmt.Sprintf("ip6.src == %s && ip6.dst == ::/0", podIP.String()))
-			if err != nil {
-				return fmt.Errorf("unable to delete logical router policy for pod IP: %s, err: %v", podIP.String(), err)
-			}
+			filterOption = fmt.Sprintf("ip6.src == %s", podIP.String())
 		} else if !utilnet.IsIPv6(podIP) && !utilnet.IsIPv6String(status.EgressIP) {
-			_, _, err := util.RunOVNNbctl("lr-policy-del", ovnClusterRouter, egressIPReroutePriority, fmt.Sprintf("ip4.src == %s && ip4.dst == 0.0.0.0/0", podIP.String()))
+			filterOption = fmt.Sprintf("ip4.src == %s", podIP.String())
+		}
+		policyIDs, err := findLRPolicyIDs(filterOption, egressIPName)
+		if err != nil {
+			return err
+		}
+		for _, policyID := range strings.Split(policyIDs, "\n") {
```

Comment:
It would make more sense to have `findLRPolicyIDs` do the split itself and return an array.

But also, it needs to check if the return value from `util.RunOVNNbctl` is `""` first and return `nil` in that case, since `strings.Split("", "\n")` will return `[]string{""}`, and so you would end up running "`ovn-nbctl remove logical_router ... policies ""`" here and then getting an error.